code/game/objects/random/food.dm

```diff
@@ -16,15 +34,17 @@
 				/obj/item/weapon/reagent_containers/food/snacks/pie = 1,\
 				/obj/item/weapon/reagent_containers/food/snacks/sandwich = 1))
 
-/obj/random/rotfood/low_chance
+/obj/random/junkfood/rotten/low_chance
 	name = "low chance spoiled food"
 	icon_state = "food-red-low"
 	spawn_nothing_percentage = 60
 
-/obj/random/rotfood/spawn_item()
+/obj/random/junkfood/rotten/spawn_item()
```

Comment:
There's a post spawn function for this, use it. 
Overriding spawn item generally shouldn't be done